File: cm-backend/src/main/java/com/clothing/management/services/UserService.java
package com.clothing.management.services;

import com.clothing.management.entities.User;
import java.util.List;

public interface UserService {

    User addUser(User user);
    List<User> findAllUsers();
    User findByUserName(String userName);
    User updateUser(User user);
    void deleteAllUsersById(List<Integer> usersId);
}

File: cm-front/src/shared/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private baseUrl: string = environment.baseUrl + "/user";
  
  constructor(private _http: HttpClient) { }

  findAllUsers() {
    return this._http.get(this.baseUrl + '/findAll');
  }

  addUser(user: any) {
    return this._http.post(this.baseUrl + '/add' , user);
  }

  updateUser(user: any) {
    return this._http.put(this.baseUrl + '/update' , user);
  }


  deleteAllUsersById(usersId: string[]) {
    return this._http.delete(this.baseUrl + '/deleteAllById/' + usersId);
  }
}
